frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -52,86 +49,94 @@ const getIconClass = (status: ClusterUpdateStatus) => {
+        &nbsp;
+        Update available
+      </div>
+      <div>
+        <Button bsStyle="primary" onClick={() => (clusterUpdateModal({cv}))}>
+          Update now
         </Button>
-        : <span>
-          {iconClass && <i className={iconClass} aria-hidden={true}></i>}
-          &nbsp;
-          {status}
-        </span>
-    }
-  </React.Fragment>;
+      </div>
+    </React.Fragment>;
+  }
+  if (status === ClusterUpdateStatus.Updating) {
+    const updatingCondition = getClusterVersionCondition(cv, ClusterVersionConditionType.Progressing, K8sResourceConditionStatus.True);
+    return <React.Fragment>
+      <div>
+        {iconClass && <i className={iconClass} aria-hidden={true}></i>}
+        &nbsp;
+        Working towards
+      </div>
+      <div>updatingCondition.message && `${updatingCondition.message}. `</div>
+      <Link to="/settings/cluster/clusteroperators?orderBy=desc&sortBy=Status">View detailed progress</Link>
+    </React.Fragment>;
+  }
+  if (status === ClusterUpdateStatus.ErrorRetrieving) {
+    return <React.Fragment>
+      <div>
+        {iconClass && <i className={iconClass} aria-hidden={true}></i>}
+        &nbsp;
+        Error retrieving
+      </div>
+      <Link to="/settings/cluster/clusteroperators?orderBy=desc&sortBy=Status">View detailed progress</Link>
+    </React.Fragment>;
+  }
+  if (status === ClusterUpdateStatus.Failing) {
+    return <React.Fragment>
+      <div>
+        {iconClass && <i className={iconClass} aria-hidden={true}></i>}
+        &nbsp;
+        {status}
+      </div>
+      <Link to="/settings/cluster/clusteroperators?orderBy=desc&sortBy=Status">View detailed progress</Link>
+    </React.Fragment>;
+  }
+  return <span>
+    {iconClass && <i className={iconClass} aria-hidden={true}></i>}
+    &nbsp;
+    Up to date
+  </span>;
 };
 
-const DesiredVersion: React.SFC<DesiredVersionProps> = ({cv}) => {
+const CurrentVersion: React.SFC<CurrentVersionProps> = ({cv}) => {
   const version = getDesiredClusterVersion(cv);
   return version
     ? <React.Fragment>{version}</React.Fragment>
     : <React.Fragment><i className="pficon pficon-warning-triangle-o" aria-hidden="true" />&nbsp;Unknown</React.Fragment>;
 };
 
+const UpdateLink: React.SFC<CurrentVersionProps> = ({cv}) => {
+  const status = getClusterUpdateStatus(cv);
+  return <React.Fragment>
+    {
+      status === ClusterUpdateStatus.ErrorRetrieving || status === ClusterUpdateStatus.Failing || status === ClusterUpdateStatus.Updating
+        ? <Button bsStyle="link" className="btn-link--no-padding" onClick={() => (clusterUpdateModal({cv}))}>Update to another version</Button>
+        : null
+    }
+  </React.Fragment>;
+};
+
+const CurrentVersionHeader: React.SFC<CurrentVersionProps> = ({cv}) => {
+  const status = getClusterUpdateStatus(cv);
+  return <React.Fragment>
+    { status === ClusterUpdateStatus.UpToDate || status === ClusterUpdateStatus.UpdatesAvailable ? 'Current Version' : 'Last Completed Version' }
```

Comment:
This is inaccurate. `CurrentVersion` does not show the last completed version as written. It shows the desired version (the one you're progressing to) if an update is in progress.

To show the last completed version, you probably need to go into `status.history`.

I'm not sure what we would show here on initial install since there is no last completed version, although typically you aren't using the console yet.

Reply:
Thanks, I dug it out of status.history. It looks like _.get(cv, 'status.history') returns an array -- do you know whether new entries are added to the end or the beginning of the array?

Right now we show "Unknown" if we don't have the version -- would you want to reuse that behavior for the last completed version?

Reply:
The newest update is first in the list.

https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L84-L92

I think we should show something like `None` if there is no completed version. (You probably wouldn't encounter this often.)

Reply:
Thanks! Added in new commit.